db/context.go

```diff
@@ -53,6 +53,11 @@ func ensureContextTxn(ctx context.Context, db transactionDB, readOnly bool) (con
 	return SetContextTxn(ctx, txn), txn, nil
```

Comment:
suggestion: Sorry I missed the PR in which this was added, would you mind noting that a new `context.Context` object may be returned from this function (in the case of implicit txn)? It is not clear from the existing documentation.

Something along the lines of:
> ... otherwise a new implicit transaction will be created, and returned within a new context object also containing the copied values from the input context.

I know this is expected with Go contexts, but it is important and thus worth being nice and explicit here IMO.

Reply:
Just adding some *context* here ;). A new `context.Context` will *always* be returned regardless of whether the implicit txn is created. Since the existing `txn` on the context is converted to an `explicityTxn` and then re-added to the context, thus creating a new context (since they are immutable).